query-engine/connectors/mongodb-query-connector/src/root_queries/mod.rs

```diff
@@ -59,19 +60,27 @@ where
     F: FnOnce() -> U + 'a,
     U: Future<Output = mongodb::error::Result<T>>,
 {
+    // TODO: build the string lazily in the Display impl so it doesn't have to be built if neither
+    // logs nor traces are enabled. This is tricky because whatever we store in the span has to be
+    // 'static, and all `QueryString` implementations aren't, so this requires some refactoring.
+    let query_string = builder.build();
```

Comment:
Ok to have this in a future PR. Is there a backlog task for this already?

Reply:
> Ok to have this in a future PR.

Yeah it was already a problem with logs (a comment above actually refers to it as well), and since we're cloning the built string and not building it again, we're not relatively adding too much overhead that didn't already exist. It should still be fixed though.

> Is there a backlog task for this already?

I don't believe so, I didn't find one, there's only a task for another todo in this function. I'll add one.

Reply:
> we're not relatively adding too much overhead

Actually if `Arc<str>` implements `tracing::Value` let's just use that so we don't add any new overhead.